core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -566,6 +579,9 @@ private[spark] class TaskSchedulerImpl(
     if (taskResultGetter != null) {
       taskResultGetter.stop()
     }
+    if (barrierCoordinator != null) {
```

Comment:
What is the case if `barrierCoordinator == null`? `barrierCoordinator` is `lazy val`.

Reply:
It shall first get inited and then get stopped. Do we have a better way for this?

Reply:
Ah, i see. How about leaving a comment?

Reply:
maybe we should not use `lazy val`, but use `var` and control the initialization ourselves.